stack/stack_hash.cpp

```diff
@@ -0,0 +1,16 @@
+#include "stack_hash.h"
+
+unsigned long djb2(const char *str) {
+    unsigned long hash = 5381;
+    int c;
+    while ((c = *str++)) {
+        hash = ((hash << 5) + hash) + c;  // hash * 33 + c
+    }
+    return hash;
+}
+
+unsigned long hash_struct(my_stack *stk) {
+    char buffer[100];
+    snprintf(buffer, sizeof(buffer), "%p%d%d", stk->data, stk->size, stk->capacity);
+    return djb2(buffer);
```

Comment:
your stack would be hacked if I will get some element from it and change to another value. you should also take this into account

Reply:
added hash for data